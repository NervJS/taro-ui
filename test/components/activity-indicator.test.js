import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'
import AtActivityIndicator from '../../.temp/components/activity-indicator/index'

describe('ActivityIndicator Snap', () => {
  it('render initial ActivityIndicator', () => {
    const component = renderToString(<AtActivityIndicator />)
    expect(component).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props size', () => {
    const component = renderToString(<AtActivityIndicator size={32} />)
    expect(component).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props color', () => {
    const component = renderToString(<AtActivityIndicator color='#13CE66' />)
    expect(component).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props content ', () => {
    const component = renderToString(
      <AtActivityIndicator content='加载中...' />
    )
    expect(component).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props mode ', () => {
    const component = renderToString(<AtActivityIndicator mode='center' />)
    expect(component).toMatchSnapshot()
  })

  it('render ActivityIndicator -- props isOpened ', () => {
    const component = renderToString(<AtActivityIndicator isOpened />)
    expect(component).toMatchSnapshot()
  })
})
