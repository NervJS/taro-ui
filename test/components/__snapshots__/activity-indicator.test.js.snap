// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityIndicator Snap render ActivityIndicator -- props color 1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:1px solid #13CE66;border-color:#13CE66 transparent transparent transparent;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:1px solid #13CE66;border-color:#13CE66 transparent transparent transparent;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:1px solid #13CE66;border-color:#13CE66 transparent transparent transparent;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;

exports[`ActivityIndicator Snap render ActivityIndicator -- props content  1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div><span class=\\"taro-text at-activity-indicator__content\\">加载中...</span></div>"`;

exports[`ActivityIndicator Snap render ActivityIndicator -- props isOpened  1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;

exports[`ActivityIndicator Snap render ActivityIndicator -- props mode  1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--center at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;

exports[`ActivityIndicator Snap render ActivityIndicator -- props size 1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:0.6827rem;height:0.6827rem;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:0.6827rem;height:0.6827rem;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:0.6827rem;height:0.6827rem;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:0.6827rem;height:0.6827rem;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;

exports[`ActivityIndicator Snap render initial ActivityIndicator 1`] = `"<div class=\\"at-activity-indicator at-activity-indicator--isopened\\"><div class=\\"at-activity-indicator__body\\"><div style=\\"width:;height:;\\" class=\\"at-loading\\"><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div><div style=\\"border:;border-color:;width:;height:;\\" class=\\"at-loading__ring\\"></div></div></div></div>"`;
